### DISCUSSION TITLE
where option of load method did not work.

### DISCUSSION THREAD

**Discussion by jays1204** [mikro-orm/mikro-orm#5564] [Q&A] [answered]
Hi

I created entities like this.

```
@Entity()
export class Book {

  @PrimaryKey()
  id: number;

  @Porperty()
  name: string;

}

@Entity()
export class Author {

  @PrimaryKey()
  id: number;

  @OneToMany(() => Book, book => book.author)
  books = new Collection<Book>(this);

}

```

When I loaded author, and I called load method with where option.
```
const author = await em.findOne(Author, 1);
const abcBooks = await author.books.load({ name: 'abc' });
console.info('abcBooks', abcBooks)
```

I expected that the query from book where name='abc' would run and abcBooks would only have data with name 'abc'.
However, the query didn't run and abcBooks contained data with name not abc.
Can you figure out why this is?
I'm using nestjs.

**Comment by B4nan** [comment, marked as answer]
`Collection.load` does not take a `where` parameter, your code won't even compile, since there is no `name` option in there, the parameter is options object (https://mikro-orm.io/api/core/class/Collection#load). So the real question is why did you ignore the fact that this won't compile (or you're not using TS at all?) and where did you come up with what you are trying, since it's not documented anywhere. 

You want `books.load({ where: { name: 'abc' }})` or `books.matching({ where: { name: 'abc' }})` (https://mikro-orm.io/docs/collections#filtering-collections).

**Reply by jays1204** [reply]
Thank you for answer. I use Typescript now. I didn't know that no where paramter in load.
I'll use matching method.

**Reply by jays1204** [reply]
I have additional questions.
You said that the where condition is ignored, is there any way to know why it is ignored?
If you look at https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/Collection.ts#L63, the options are using the InitCollectionOptions type.
The InitCollectionOptions type extends EntityLoader if you look at https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/Collection.ts#L477.
EntityLoader has a where defined as seen in https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/EntityLoader.ts#L35, so can you see why the where is being ignored?

**Reply by B4nan** [reply]
I said there is no `where` parameter, you need to use the `where` option, not put the whole `where` condition instead of the options parameter.

**Reply by jays1204** [reply]
I understand it. Thank you!
